### PR TITLE
fix: Event handlers firing twice on non-interacting root display objects

### PR DESCRIPTION
**Pull request by CatchABus**
##### Description of change
Right now, property event handlers like `onclick` can be triggered for non-intaractive display objects. In the case of the root display object, listeners will be called twice.
I added 2 unit tests that seem to catch the case.

This PR is a continuation of #9166
Note: This bug seems similar to #10441 but it's not the same defect.

##### Pre-Merge Checklist
- [x] Tests and/or benchmarks are included
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -117,7 +117,7 @@
     "ts-jest": "^26.0.0",
     "ts-node": "^9.0.0",
     "tsconfig-paths": "^3.10.1",
-    "typescript": "^5.0.4",
+    "typescript": "~5.3.3",
```
Comment (bigtimebuddy) [inline package.json:120]: Could you explain this change?

Comment (bigtimebuddy) [reply package.json:120]: I'd like to consider this as a separate PR. It's outside the scope of what you're trying to do here and I think for our case, that ESLint warning is benign.

Comment (CatchABus) [reply package.json]: Done! Thank you very much for pointing out! :smile:

**Review by bigtimebuddy — APPROVED** [review]
Nice test

**Pull request merged**